core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -164,11 +164,10 @@ Optional<TransactionResult> read(@Nullable Snapshot.Key key, Get get) throws Cru
     }
 
     if (result.isPresent() || get.getConjunctions().isEmpty()) {
-      // Keep the read set latest to create before image by using the latest record (result)
-      // because another conflicting transaction might have updated the record after this
-      // transaction read it first. However, we update it only if a get operation has no
```

Comment:
Sorry, my bad. This part was a completely wrong idea. Probably, I was trying to avoid aborts as much as possible, but paid little attention to the case for reading a record multiple times. I need to think about how we can avoid these kinds of concurrency bugs…